packages/components/src/internal/app/utils.ts

```diff
@@ -171,6 +173,17 @@ export function hasProductProjects(moduleContext?: ModuleContext): boolean {
     return resolveModuleContext(moduleContext)?.query?.hasProductProjects === true;
 }
 
+export function setProductProjects(moduleContext: ModuleContext, hasProductProjects: boolean): ModuleContext {
```

Comment:
I wonder if this function should not take moduleContext as param, but rather get moduleContext from getServerContext().moduleContext. Having moduleContext as param allows overriding the entire moduleContext (any fake moduleContext can be constructed and passed in). But I think limiting updates to hasProductProjects only is a better way to go.